app/classifier/task-nav.jsx

```diff
@@ -0,0 +1,216 @@
+
+    classification.annotations.pop();
+    classification.update('annotations');
+
+    if (workflow.configuration.persist_annotations) {
+      CacheClassification.update(lastAnnotation);
+    }
+  }
+
+  warningToggleOn() {
+    if (!this.props.workflow.configuration.persist_annotations) {
+      this.setState({ backButtonWarning: true });
+    }
+  }
+
+  warningToggleOff() {
+    if (!this.props.workflow.configuration.persist_annotations) {
+      this.setState({ backButtonWarning: false });
+    }
+  }
+
+  render() {
+    if (!this.props.annotation) {
+      return null;
+    }
+    const task = this.props.task ? this.props.task : this.props.workflow.tasks[this.props.workflow.first_task];
+
+    const disableTalk = this.props.classification.metadata.subject_flagged;
+    const visibleTasks = Object.keys(this.props.workflow.tasks).filter(key => this.props.workflow.tasks[key].type !== 'shortcut');
+    const TaskComponent = tasks[task.type];
+
+    // Should we disable the "Back" button?
+    const onFirstAnnotation = (this.props.classification.annotations.indexOf(this.props.annotation) === 0);
+
+    // Should we disable the "Next" or "Done" buttons?
+    let waitingForAnswer;
+    if (TaskComponent && TaskComponent.isAnnotationComplete) {
+      waitingForAnswer = !this.props.annotation.shortcut && !TaskComponent.isAnnotationComplete(task, this.props.annotation, this.props.workflow);
+    }
+
+    // Each answer of a single-answer task can have its own `next` key to override the task's.
+    let nextTaskKey;
+    if (TaskComponent === tasks.single) {
+      const currentAnswer = task.answers[this.props.annotation.value];
+      nextTaskKey = currentAnswer ? currentAnswer.next : '';
+    } else {
+      nextTaskKey = task.next;
+    }
+
+    if (nextTaskKey && !this.props.workflow.tasks[nextTaskKey]) {
+      nextTaskKey = '';
+    }
+
+    // TODO: Actually disable things that should be.
+    // For now we'll just make them non-mousable.
+    const disabledStyle = {
+      opacity: 0.5,
+      pointerEvents: 'none'
+    };
+
+    return (
+      <div>
+        <nav className="task-nav">
+          {(visibleTasks.length > 1) &&
+            <button
+              type="button"
+              className="back minor-button"
+              disabled={onFirstAnnotation}
+              onClick={this.destroyCurrentAnnotation}
+              onMouseEnter={this.warningToggleOn}
+              onFocus={this.warningToggleOn}
+              onMouseLeave={this.warningToggleOff}
+              onBlur={this.warningToggleOff}
+            >
+              Back
+            </button>}
+          {(!nextTaskKey && this.props.workflow.configuration.hide_classification_summaries && this.props.project && !disableTalk) &&
+            <Link
+              onClick={this.completeClassification}
+              to={`/projects/${this.props.project.slug}/talk/subjects/${this.props.subject.id}`}
```

Comment:
`this.props.subject.id` is throwing `TypeError: Cannot read property 'id' of undefined` and the task nav buttons aren't rendering:

https://classifier-task-es6.pfe-preview.zooniverse.org/projects/srallen086/workflow-assignment-testing/classify